### PR TITLE
I am checking in code that processes data appends to tabular form

### PR DESCRIPTION
**Pull request by datasciantle**
Hi Folks- I spoke to a few of you about the need to have some etl boiler plate for Append.  Please let me know if you think this script is appropriate for most teams to get started.

### CODE REVIEW

File: append-etl/etl-append-data-to-tables.scala [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+// Define UDFs
+
+// UDF to split the 'ATTRIBUTES' column into an array of svkeys
+val splitSvkeys = udf((attributes: String) => attributes.split(",\\s*"))
+
+
+// Read in the data assets
+val dataAppend = spark.read
+  .option("header", "true")
```
Comment (nathan-resonate) [inline append-etl/etl-append-data-to-tables.scala]: These files do not have a header row according to the spec (https://support.resonate.com/hc/en-us/articles/11148661289239-Data-Append-Deliverables-HEMs-or-MAIDs-Only), and based on the sample file I'm looking at (s3://resonate-data-append-dev/client=NewsBreak/date=20230130/append_results/part-00000-2c0aa47f-c006-4e1a-8620-f3731906bdc0-c000.csv).

File: append-etl/etl-append-data-to-tables.scala [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+val allAsMultiSelectDF = dataAppendWithSvkeys
+  .select(col("ID"), explode(col("trueSvkeys")).as("survey_value_key"))
+  .join(focalTaxonomy, "survey_value_key")
+  .groupBy("ID")
+  .pivot("attributeString", allAttributes)
+  .agg(count("survey_value_key"))
+  .na.fill(0)
+
+// Save the processed DataFrames
+singleSelectDF.write.mode("overwrite").save(params.outDir + "singleSelectDF")
```
Comment (nathan-resonate) [inline append-etl/etl-append-data-to-tables.scala:124]: In my testing, it looked like the values for the attribute key columns in this dataset were attribute keys not survey value keys.

File: append-etl/etl-append-data-to-tables.scala (hunk 1 above)
Comment (datasciantle) [reply append-etl/etl-append-data-to-tables.scala]: Thanks for sharing the specs - files I have for testing included a header.  Added a line to read in without header, set col names, and drop rows where ID = ID in event there's a header.

File: append-etl/etl-append-data-to-tables.scala (hunk 2 above)
Comment (datasciantle) [reply append-etl/etl-append-data-to-tables.scala:124]: I've fixed this

**Review by nathan-resonate — APPROVED** [review]

**Pull request merged**